### ISSUE TITLE
Is it possible to use javascript on the server to create custom html **before** it is sent over the wire?

### ISSUE THREAD

**Issue by RobSisson** [hotwired/turbo#807]
Hi Everyone,

Mad love to the community around this package, its absolutely unreal. 

I'm still getting stuck into it, through the [django port ](https://github.com/hotwire-django/turbo-django) however I've got a question which I don't think (?) is django specific - tell me if otherwise!

Is it possible to pre-construct the html in the backend with javascript, and then send that html over the wire?
or if not, 
Is it possible to run inline javascript which is sent over the wire as html, after it has arrived?

My use case is constructing a visualisation with a JS library, which I want to be seemlessly updated without the client having to rebuild/refresh it each time. 

I've got it so I can populate the turbo frame with the visualisation on initial load (using inline JS for now, but I'm confident this will work as expected, as the JS there is functioning as expected), however when I construct a html file with the same content in, the javascript is sent across as plain html. Therefore, the visualisation is not constructed.

Basically I want to be able to run the JS which constructs the visualisation on the server, then just send across the updated html; reducing the client load. 

Hopefully that makes sense, let me know if you need further details or examples,

Thanks, Rob

**Comment by florimondmanca** [comment]
Hi,

(Side note - Discussions have not been enabled on this repo yet, but maybe they should as this issue is a prime example of something that doesn’t require immediate maintainer attention while posing interesting usage questions.)

Answering this as a by-passer… I would say it heavily depends on the capability of the JS mapping library you’re using. What JS map library are you using? You seem to imply that it supports showing a map from an HTML representation, is that correct?

From personal wanderings on the HTMX forums (HTMX is a library similar to but less opinionated than Turbo), it is said that integration between heavy-JS client side components and a hypermedia-focused tool like Turbo is best handled via client-side events. https://htmx.org/essays/hypermedia-friendly-scripting/#events

When handling user inputs, this puts a condition on the JS library: it should expose a rich event model. Then you can hook onto those events to trigger visits. You might need a bit of scripting (Stimulus can help) to bind calls to Turbo.visit() to those events.

OTOH, when receiving updates from the server, which is your initial question, IMO the JS library should be able to update the map by reading an HTML representation read from the DOM, and this operation should be triggerable through a DOM event. If the « read the DOM » part isn’t built-in but the library does expose some kind of update/sync mechanism, you could implement yourself with a bit of scripting (again, Stimulus could help).
